indexing-service/src/main/java/org/apache/druid/indexing/overlord/IndexerMetadataStorageAdapter.java

```diff
@@ -58,11 +59,15 @@ public int deletePendingSegments(String dataSource, Interval deleteInterval)
         DateTimes.MAX
     );
 
-    Preconditions.checkArgument(
-        !deleteInterval.overlaps(activeTaskInterval),
-        "Cannot delete pendingSegments because there is at least one active task created at %s",
-        activeTaskInterval.getStart()
-    );
+    if (deleteInterval.overlaps(activeTaskInterval)) {
```

Comment:
Since we are touching this code anyway, I wonder if it would be helpful to include the Task IDs too.

Reply:
Yep, good point. I've included the earliest created task id in the exception message.